### PR TITLE
Allow multi-replica snapshot/clone

### PR DESCRIPTION
**Pull request by hrudaya21**
1. Remove the checks in the code to support multi-replica snapshot and restore from control-plane.
2. Added and modified existing BDD cases to support multi-replica snapshots and restore

### CODE REVIEW

**Review by tiagolobocastro — COMMENTED** [review]
Split into 2 commits please, disable and testing
Also can we extend the tests with rebuilds where we basically will rebuild a new replica without the snapshot?
Also add restores for this case above as well as for when replica is only offline, and not rebuilt yet

**Comment by tiagolobocastro** [comment]
And please add description to the PR comment

**Comment by hrudaya21** [comment]
> Split into 2 commits please, disable and testing Also can we extend the tests with rebuilds where we basically will rebuild a new replica without the snapshot? Also add restores for this case above as well as for when replica is only offline, and not rebuilt yet

fixed the review comments.

File: tests/bdd/features/snapshot/restore/create.feature [hunk 1]
```diff
@@ -27,6 +31,9 @@ Feature: Create Volume From Snapshot
     And we create another snapshot of the restored volume
     Then the snapshot 2 allocated size should be 4MiB
     And the restored volume total allocation size should be 8MiB
+    Examples:
+    | repl_count  |
+    | 1           |
```
Comment (tiagolobocastro) [inline tests/bdd/features/snapshot/restore/create.feature:36]: missing 3?

File: tests/bdd/features/snapshot/restore/create.feature [hunk 2]
```diff
@@ -36,3 +43,14 @@ Feature: Create Volume From Snapshot
...
+  Scenario Outline: Create a new volume as a snapshot restore with one node down
+    When we have a node down
+    And a request to create a new volume with the snapshot as its source
+    Then the request should fail with PreconditionFailed
+    When we bring the node back up
+    And we create a new volume with <repl_count> replicas using snapshot as its source
+    Then the request should succeed
+    Examples:
+    | repl_count |
+    | 3          |
```
Comment (tiagolobocastro) [inline tests/bdd/features/snapshot/restore/create.feature:58]: missing 1?

**Review by tiagolobocastro — COMMENTED** [review]
One final question, we need the validation thing to make sure there's no rebuilds on older version, will this come after this is merged?

File: tests/bdd/features/snapshot/restore/create.feature (hunk 1 above)
Comment (hrudaya21) [reply tests/bdd/features/snapshot/restore/create.feature:36]: fixed

File: tests/bdd/features/snapshot/restore/create.feature (hunk 2 above)
Comment (hrudaya21) [reply tests/bdd/features/snapshot/restore/create.feature:58]: fixed

**Comment by hrudaya21** [comment]
> One final question, we need the validation thing to make sure there's no rebuilds on older version, will this come after this is merged?

As per the last discussion, We need to have a capability feature in place to handle it, which can be a seperate story. For  this feature, We need to update the documentation to restrict creating snapshot during upgrade.

**Review by tiagolobocastro — APPROVED** [review]

**Review by sinhaashish — APPROVED** [review]

**Comment by hrudaya21** [comment]
bors try

**Comment by hrudaya21** [comment]
bors merge

**Comment by hrudaya21** [comment]
bors merge

**Pull request merged**